engine/script/src/script_timer.cpp

```diff
@@ -198,21 +174,64 @@ namespace dmScript
         delete timer_world;
     }
 
+    static Timer* GetTimerFromIndex(HTimerWorld timer_world, uint16_t index)
+    {
+        assert(timer_world != 0x0);
+        if (!timer_world->m_Instances.Contains(index))
+        {
+            return 0;
+        }
+        return &timer_world->m_Timers.Get(index);
+    }
+
+    static Timer* GetTimerFromHandle(HTimerWorld timer_world, HTimer handle)
+    {
+        uint16_t index = GetIndexFromHandle(handle);
+        Timer* timer = GetTimerFromIndex(timer_world, index);
+        if (!timer)
+            return 0;
+        if (timer->m_Handle != handle)
+            return 0; // Stale handle
+        return timer;
+    }
+
+    static uint32_t CopyIndices(dmSet<uint16_t>& src, dmArray<uint16_t>& tgt)
+    {
+        uint32_t size = src.Size();
+        if (size > tgt.Capacity())
+        {
+            tgt.SetCapacity(size);
+        }
+        tgt.SetSize(size);
+        for (uint32_t i = 0; i < size; ++i)
+        {
+            tgt[i] = src[i];
+        }
+        return size;
+    }
```

Comment:
Since the object pool may change while iterating on it, we need to make a copy of the valid instances.